ec/ecresource/deploymentresource/traffic_filters.go

```diff
@@ -0,0 +1,95 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package deploymentresource
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+const trafficFilterPrivateStateKey = "traffic_filters"
+
+type PrivateDataStore interface {
+	SetKey(context.Context, string, []byte) diag.Diagnostics
+	GetKey(context.Context, string) ([]byte, diag.Diagnostics)
+}
+
+func CaptureManagedFiltersForCreate(ctx context.Context, config tfsdk.Config, privateData PrivateDataStore) diag.Diagnostics {
+	var filters []string
+	var diags = config.GetAttribute(ctx, path.Empty().AtName("traffic_filter"), &filters)
+	if diags.HasError() {
+		return diags
+	}
+
+	data, err := json.Marshal(filters)
+	if err != nil {
+		diags.AddError("unable to prepare traffic filter private state", err.Error())
+		return diags
+	}
+
+	diags = privateData.SetKey(ctx, trafficFilterPrivateStateKey, data)
+	return diags
+}
+
+func CaptureManagedFiltersForUpdate(ctx context.Context, config tfsdk.Config, privateData PrivateDataStore) diag.Diagnostics {
+	var filters []string
+	var diags = config.GetAttribute(ctx, path.Empty().AtName("traffic_filter"), &filters)
+	if diags.HasError() {
+		return diags
+	}
+
+	data, err := json.Marshal(filters)
+	if err != nil {
+		diags.AddError("unable to prepare traffic filter private state", err.Error())
+		return diags
+	}
+
+	privateData.SetKey(ctx, trafficFilterPrivateStateKey, data)
+	return diags
+}
+
+func GetManagedFilters(ctx context.Context, req resource.UpdateRequest, privateData PrivateDataStore) ([]string, diag.Diagnostics) {
+	filterBytes, diags := privateData.GetKey(ctx, trafficFilterPrivateStateKey)
+	if diags.HasError() {
+		return []string{}, diags
+	}
+
+	var filters []string
+	if filterBytes == nil {
+		var filtersTF types.Set
+		diags = req.State.GetAttribute(ctx, path.Empty().AtName("traffic_filter"), &filters)
```

Comment:
diags = req.State.GetAttribute(ctx, path.Root("traffic_filter"), &filters)
```suggestion
		diags = req.State.GetAttribute(ctx, path.Empty().AtName("traffic_filter"), &filters)
```